evalml/utils/gen_utils.py

```diff
@@ -556,23 +564,67 @@ def are_ts_parameters_valid_for_split(
         forecast_horizon (int): forecast_horizon value.
         n_obs (int): Number of observations in the dataset.
         n_splits (int): Number of cross validation splits.
+        problem_type (str or ProblemTypes): Problem type.
+        y (pd.Series): Target data.
 
     Returns:
         TsParameterValidationResult - named tuple with four fields
             is_valid (bool): True if parameters are valid.
             msg (str): Contains error message to display. Empty if is_valid.
             smallest_split_size (int): Smallest split size given n_obs and n_splits.
             max_window_size (int): Max window size given gap, max_delay, forecast_horizon.
+            invalid_split (list): All the invalid splits as a result of inadequate target class representation.
     """
+    from evalml.problem_types import ProblemTypes, handle_problem_types
+
     split_size = n_obs // (n_splits + 1)
     window_size = gap + max_delay + forecast_horizon
     msg = ""
+
+    is_split_less_than_window = False
+    is_split_one_class = False
+    invalid_splits = []
+
     if split_size <= window_size:
+        is_split_less_than_window = True
+    if y is not None and handle_problem_types(problem_type) in [
```

Comment:
@ParthivNaresh Quick question: at OH we discussed that this check only needed to apply to the first fold. Why the change here? Can we simplify this by only checking the first fold?